server/src/main/java/io/crate/types/StringType.java

```diff
@@ -79,77 +80,87 @@ public class StringType extends DataType<String> implements Streamer<String> {
         */
     public static final int POSITION_INCREMENT_GAP = 100;
 
-    private static final StorageSupport<Object> STORAGE = new StorageSupport<>(
-        true,
-        true,
-        new EqQuery<Object>() {
+    protected static final class StringEqQuery implements EqQuery<Object> {
 
-            @Override
-            public Query termQuery(String field, Object value, boolean hasDocValues, boolean isIndexed) {
-                if (isIndexed) {
-                    return new TermQuery(new Term(field, BytesRefs.toBytesRef(value)));
-                }
-                if (hasDocValues) {
-                    return SortedSetDocValuesField.newSlowExactQuery(field, BytesRefs.toBytesRef(value));
-                }
-                return null;
-            }
+        private final UnaryOperator<Object> preProcess;
 
-            @Override
-            public Query rangeQuery(String field,
-                                    Object lowerTerm,
-                                    Object upperTerm,
-                                    boolean includeLower,
-                                    boolean includeUpper,
-                                    boolean hasDocValues,
-                                    boolean isIndexed) {
-                if (isIndexed) {
-                    return new TermRangeQuery(
-                        field,
-                        BytesRefs.toBytesRef(lowerTerm),
-                        BytesRefs.toBytesRef(upperTerm),
-                        includeLower,
-                        includeUpper
-                    );
-                }
-                if (hasDocValues) {
-                    return SortedSetDocValuesField.newSlowRangeQuery(
-                        field,
-                        BytesRefs.toBytesRef(lowerTerm),
-                        BytesRefs.toBytesRef(upperTerm),
-                        includeLower,
-                        includeUpper
-                    );
-                }
-                return null;
+        StringEqQuery(UnaryOperator<Object> preprocess) {
```

Comment:
Do you know why StorageSupport is typed `StorageSupport<Object>` instead of `<String>`?

A bit unrelated to this PR but would be nice to have a comment that explains that, or even change it to String if possible

Reply:
Oh, based on the tests it's the search after optimization - so yes a comment about that would be good

Reply:
yep, adding comment